src/lib/RingBuffer.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import {Arrays} from "@openzeppelin/contracts/utils/Arrays.sol";
+import {Panic} from "@openzeppelin/contracts/utils/Panic.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+library RingBuffer {
+    using Arrays for bytes32[];
+
+    struct Checkpoint {
+        /// @dev Maps checkpoints to an always increasing index to ensure a checkpoint belongs
+        /// to the current ring (e.g. proving between 3 and 5 will leave indexes 4 and 5 untouched)
+        uint256 index;
+        /// @dev May be anything that describes the rollup state at a given data feed publication (e.g. a state root)
+        bytes32 value;
+    }
+
+    struct CheckpointBuffer {
+        /// @dev Represents the latest overall index (i.e. the index if the buffer didn't wrap around)
+        uint256 _lastIndex;
```

Comment:
shall we remove leading "_"?

Reply:
I use to prefix with `_` when the developers are not supposed to access the value directly (as in saying that it's private). No strong opinion here so I'll update.